src/helpers/checkBinary.ts

```diff
@@ -1,10 +1,6 @@
 const checkBinary = (value: string) => {
   const isBinary = value.split("").every((n) => n === "0" || n === "1");
-  if (Number(value) < 0 || !isBinary) {
-    return false;
-  } else {
-    return value;
-  }
+  return !(Number(value) < 0 || !isBinary);
```

Comment:
What is `Number(value)` actually doing for you here? It seems like the statement on line 2 gives you all the validation you need _and_ you pay the time complexity of `n` cost regardless of the outcome of `Number(value)`

Unless I'm missing something you can just return the result of `isBinary`

```
return value.split("").every((n) => n === "0" || n === "1");
```

also the bool flip `!(!isBinary)` is dirty to read

Reply:
That's true. I was doing a redundant check there, in fact, I don't need to check for a positive number. 
Good call.